app/helpers/application_helper/button/cloud_volume_new.rb

```diff
@@ -0,0 +1,18 @@
+class ApplicationHelper::Button::CloudVolumeNew < ApplicationHelper::Button::Basic
+  def calculate_properties
+    super
+    if disabled?
+      self[:title] = _("No cloud providers support creating cloud volumes.")
+    end
+  end
+
+  def disabled?
+    ems_clouds = EmsCloud.all
+    # if any connected provider supports this action,
+    # we can enable the button.
+    ems_clouds.each do |ems|
+      return false if CloudVolume.class_by_ems(ems).supports_create?
+    end
+    true
```

Comment:
I don't know if the above part can be improved performance-wise, but perhaps, we could make the whole
thing better by moving this logic right into the cloud volume create form. So that the add button will always be
visible in the ui (i.e. no complex queries when rendering the cloud volume list) and in the form we'd have a banner
(or whatever) saying something along the lines of "There's no cloud provider supporting cloud volume creation available".